vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -458,11 +463,15 @@ window.Vaadin.Flow.gridConnector = {
         if (parentCache && parentCache.itemkeyCaches && parentCache.itemkeyCaches[parentKey]) {
           delete parentCache.itemkeyCaches[parentKey];
         }
-        if (parentCache && parentCache.itemCaches && parentCache.itemCaches[parentKey]) {
-          delete parentCache.itemCaches[parentKey];
+        if (parentCache && parentCache.itemCaches) {
+          const keys = Object.keys(parentCache.itemCaches);
```

Comment:
The change works well and overall lgtm. Any idea why the tests aren't passing (would have approved this otherwise)?

nit: Logic here took me a while to understand. To improve readability it could be changed to:
```
Object.keys(parentCache.itemCaches)
        .filter(idx => parentCache.items[idx].key === parentKey)
        .forEach(idx => delete parentCache.itemCaches[idx]);
```

Reply:
I changed this. I don't know why the tests are not passing.
On my computer, I've got some errors on the grid tests but not the same.
I tried treegridhuge test, and (if I removed ObjectSizeCalculator to try it) I don't have this error. I try it manually and it seems ok. I'll check the result after this modification.

Reply:
I think the wrapping if-statement is still needed?

Reply:
Yes, it still needed, even if no test is checking this. (never push code before morning coffee).
I increased the timeout for the test to test if it's working but slower than before.